packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -595,10 +604,27 @@ exports.createAssetNodes = ({
   defaultLocale,
   locales,
   space,
+  assetImagesFallbackOverride,
 }) => {
   const createNodePromises = []
   locales.forEach(locale => {
-    const localesFallback = buildFallbackChain(locales)
+    const localesFallback = (() => {
+      const fallbacks = buildFallbackChain(locales)
+
+      if (
+        assetImagesFallbackOverride &&
+        assetItem.fields.file[defaultLocale].details.image
+      ) {
+        return locales.reduce((accumulator, current) => {
+          return {
+            ...accumulator,
+            [current.code]: fallbacks[current.code] || defaultLocale,
+          }
+        }, {})
+      }
+
+      return fallbacks
+    })()
```

Comment:
Can we please do this without a local iife? Nor with a reduce with dynamic property literal spread etc.

I would argue something like this is simpler to follow and maintain;

```suggestion
    const fallbacks = buildFallbackChain(locales)
    let localesFallback = fallbacks
    if (
      assetImagesFallbackOverride &&
      assetItem.fields.file[defaultLocale].details.image
    ) {
      localesFallback = {}
      locales.forEach(locale => localesFallback[locale.code] = fallbacks[current.code] || defaultLocale)
    }
```

Reply:
Furthermore;

The `buildFallbackChain` is a static function. Please cache the generated `fallbacks` array outside of the `locales` loop. (_I realize you didn't put it in the loop, but since you're changing the code, you may as well put it outside the loop...?_)

Considering the above is static, it seems to me this new loop you've added also uses static information that is not changed inside the loop, so please generate the object once outside the loop, if the flag asks for it.

Reply:
I wonder if this is nicer for the new loop. Probably not.
```js
localesFallback = {...fallback}
locales.forEach(({code}) => localesFallback[code] ?? (fallbacks[code] = defaultLocale)
```